tests/test_utils.py

```diff
@@ -1,35 +1,89 @@
-# import pytest
-# from agoradatatools.etl import utils
-# from synapseclient import Synapse
+import pytest
+from agoradatatools.etl import utils
+from unittest.mock import patch
+import synapseclient
+import yaml
+import io
+import builtins
 
-# file_object = {
-#     "id": "syn25838546",
-#     "format": "table",
-#     "final_filename": "teams",
-#     "provenance": [],
-#     "destination": "syn25871921"
-# }
 
+class TestLoginToSynapse:
+    @pytest.fixture(scope="function", autouse=True)
+    def setup_method(self, syn):
+        self.patch_synapseclient = patch.object(
+            synapseclient, "Synapse", return_value=syn
+        ).start()
+        self.patch_syn_login = patch.object(syn, "login", return_value=syn).start()
 
-# def test_login():
-#     assert type(utils._login_to_synapse()) is Synapse
+    def teardown_method(self):
+        self.patch_synapseclient.stop()
+        self.patch_syn_login.stop()
 
+    def test_login_with_token(self):
+        utils._login_to_synapse(authtoken="my_auth_token")
+        self.patch_syn_login.assert_called_once_with(authToken="my_auth_token")
 
-# def test_yaml():
-#     # tests if a valid file renders a list
-#     assert type(utils._get_config()) is list
+    def test_login_no_token(self):
+        utils._login_to_synapse(authtoken=None)
+        self.patch_syn_login.assert_called_once_with()
 
-#     # tests if a bad file will
-#     with pytest.raises(SystemExit) as err:
-#         utils._get_config(config_path="./tests/test_assets/bad_config.yaml")
-#     assert err.type == SystemExit
-#     assert err.value.code == 9
 
-#     with pytest.raises(SystemExit) as err:
-#         utils._get_config(config_path="./tests/test_assets/bad_config.yam")
-#     assert err.type == SystemExit
-#     assert err.value.code == 2
+def test_get_config_with_invalid_file_path():
+    with pytest.raises(FileNotFoundError, match="File not found. *"):
+        utils._get_config(config_path="this/is/a/bad/path")
 
 
-# if __name__ == "__main__":
-#     pytest.main()
+def test_get_config_with_parser_error():
+    with pytest.raises(
+        yaml.parser.ParserError, match="YAML file unable to be parsed. *"
+    ):
+        utils._get_config(config_path="./tests/test_assets/bad_config_parsing.yaml")
+
+
+def test_get_config_with_scanner_error():
+    with pytest.raises(
+        yaml.scanner.ScannerError, match="YAML file unable to be scanned. *"
+    ):
+        utils._get_config(config_path="./tests/test_assets/bad_config_scanning.yaml")
+
+
+# class TestGetConfigWithAndWithoutPath:
```

Comment:
For simplicity sake, I suggest us to just not mock the `open` module here.   This way we can actually assert that the config file is being read by checking a value within the config that can uniquely identify the config file.

Reply:
done in latest commit